expr/functions/nonNegativeDerivative/function.go

```diff
@@ -85,9 +85,10 @@ func (f *nonNegativeDerivative) Do(ctx context.Context, e parser.Expr, from, unt
 			name = "nonNegativeDerivative(" + a.Name + ")"
 		}
 
-		r := *a
+		r := a.CopyLink()
 		r.Name = name
 		r.Values = make([]float64, len(a.Values))
+		r.Tags["nonNegativeDerivative"] = "1"
```

Comment:
Need copy-on-write for tags. May be encapsylate logic with special Copy function ?

Reply:
Just a thought for some distant bright future: I wanted to use semgrep to automatically check that tags are copied when needed. AFAIR it should be able to warn when some one is not using CopyLink but modifying tags. Maybe now it would be worth investing time into that.

Reply:
It's my mistake, CopyLink copy tags map. Perhaps I chose not very good names.